assets/tests/TaskSchemaValidationService.js

```diff
@@ -0,0 +1,240 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for validating data put into a task schema
+ * within a question.
+ */
+
+tie.factory('TaskSchemaValidationService', ['SYSTEM_CODE',
+  function(SYSTEM_CODE) {
+    var AUXILIARY_CODE_CLASS_NAME = 'AuxiliaryCode.';
+    var SYSTEM_CODE_CLASS_NAME = 'System.';
+    var VALID_PREFIXES = [
+      AUXILIARY_CODE_CLASS_NAME, SYSTEM_CODE_CLASS_NAME];
+    var starterCode;
+    var auxiliaryCode;
+    // TODO(eyurko): Update this once we support nonlinear runtimes.
+    var ALLOWED_RUNTIMES = ['linear'];
+    // TODO(eyurko): Add check that definition occurs at line start.
+    var _checkIfFunctionExistsInCode = function(functionName, code) {
+      var functionDefinition = 'def ' + functionName;
+      return code.indexOf(functionDefinition) !== -1;
+    };
+    // TODO(eyurko): Add check that function exists in specified class.
+    var _checkIfFunctionExistsInClass = function(functionName, 
+      className, code) {
+      if (functionName.startsWith(className)) {
+        var strippedFunctionName = functionName.substring(
+          className.length);
+        return _checkIfFunctionExistsInCode(
+          strippedFunctionName, code);
+      }
+      return false;
+    };
+    return {
+      getAllVerificationFunctions: function() {
+        var functions = [];
+        for (var func in this) {
+          if (this.hasOwnProperty(func) && 
+              angular.isFunction(this[func]) &&
+              !(/functions/i).test(func) &&
+              func.toString().startsWith('verify')) {
+            functions.push(this[func]);
+          }
+        }
+        return functions;
+      },
+      init: function(sCode, aCode) {
```

Comment:
Try to avoid abbreviations. Call the internal variables _starterCode and _auxiliaryCode, and use starterCode and auxiliaryCode for the args here.